proxygen_cli/lib/credentials.py

```diff
@@ -29,8 +38,8 @@ class Credentials(BaseSettings):
     key_id: Optional[str] = None
     client_id: str
     client_secret: str = None
```

Comment:
out of curiosity, what happens to the value if you leave this blank when running credential set-up? does it get set as an empty string, or stay as None?

i think it might be best to set this to an Optional[str] as you alluded to in the ticket comments, but i'm not sure how type enforcing actually works in the CLI's code

Reply:
So the type checking is mainly for the validation stuff. You can't actually leave it blank in the CLI as it will just keep asking you to enter a client_secret. This is the behaviour of click. I have changed it so that it is Optional[str] now and change the CLI prompt so that the user would be able to keep this as an empty string now.